lib/convert.js

```diff
@@ -1,60 +1,142 @@
+ * @param {String} gltfPath Path of the converted glTF file.
+ * @param {Object} [options] An object with the following properties:
+ * @param {Boolean} [options.binary=false] Save as binary glTF.
+ * @param {Boolean} [options.separate=false] Writes out separate geometry data files, shader files, and textures instead of embedding them in the glTF.
+ * @param {Boolean} [options.separateTextures=false] Write out separate textures only.
+ * @param {Boolean} [options.compress=false] Quantize positions, compress texture coordinates, and oct-encode normals.
+ * @param {Boolean} [options.optimize=false] Use the optimization stages in the glTF pipeline.
+ * @param {Boolean} [options.optimizeForCesium=false] Optimize the glTF for Cesium by using the sun as a default light source.
+ * @param {Boolean} [options.generateNormals=false] Generate normals if they are missing.
+ * @param {Boolean} [options.ao=false] Apply ambient occlusion to the converted model.
+ * @param {Boolean} [options.kmc=false] Output glTF with the KHR_materials_common extension.
+ * @param {Boolean} [options.textureCompressionOptions] Options sent to the compressTextures stage of gltf-pipeline.
+ * @param {Boolean} [options.bypassPipeline=false] Bypass the gltf-pipeline for debugging purposes. This option overrides many of the options above and will save the glTF with the KHR_materials_common extension.
+ * @param {Boolean} [options.hasTransparency=false] Do a more exhaustive check for texture transparency by looking at the alpha channel of each pixel.
+ * @param {Boolean} [options.secure=false] Prevent the converter from reading image or mtl files outside of the input obj directory.
+ * @param {Logger} [options.logger] A callback function for handling logged messages. Defaults to console.log.
+ */
+function convert(objPath, gltfPath, options) {
+    return new Promise(function(resolve, reject) {
+        options = defaultValue(options, {});
+        var binary = defaultValue(options.binary, false);
+        var separate = defaultValue(options.separate, false);
+        var separateTextures = defaultValue(options.separateTextures, false) || separate;
+        var compress = defaultValue(options.compress, false);
+        var optimize = defaultValue(options.optimize, false);
+        var optimizeForCesium = defaultValue(options.optimizeForCesium, false);
+        var generateNormals = defaultValue(options.generateNormals, false);
+        var ao = defaultValue(options.ao, false);
+        var kmc = defaultValue(options.kmc, false);
+        var textureCompressionOptions = options.textureCompressionOptions;
+        var bypassPipeline = defaultValue(options.bypassPipeline, false);
+        var logger = defaultValue(options.logger, defaultLogger);
+        options.logger = logger;
+        options.hasTransparency = defaultValue(options.hasTransparency, false);
+        options.secure = defaultValue(options.secure, false);
+
+        if (!defined(objPath)) {
```

Comment:
All of these DeveloperError checks should happen at the top of the function outside of the Promise block.  Developer errors should always be thrown as early as possible, not as part of Promise rejection.